pkgs/by-name/ti/tigerjython/package.nix

```
@@ -0,0 +1,87 @@
+{ stdenv
+, lib
+, fetchurl
+, makeWrapper
+, makeDesktopItem
+, copyDesktopItems
+, jre
+}:
+let
+  icon = fetchurl {
+    # In case tigerjython/tjinstall becomes unavailable, use this url - see comment for src in MkDerivation
+    #url = "https://web.archive.org/web/20240623120114/https://raw.githubusercontent.com/tigerjython/tjinstall/master/tjlogo64.png";
+    url = "https://raw.githubusercontent.com/tigerjython/tjinstall/611c56d4e765731883656a5c4b71209d72b5ab74/tjlogo64.png";
+    hash = "sha256-tw3uDWLtcMHYmN6JGsEvVKLgI09v5DF27V2+OF9Z5tA=";
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "tigerjython";
+  version = "2.39";
+
+  # we cache potentially unstable upstream input (.tar.gz file) via https://web.archive.org - this is a common procedure in Nixpkgs
+  src = fetchurl {
+    url = "http://web.archive.org/web/20240119124245/https://tigerjython.ch/user/pages/download/TigerJython.tar.gz";
+    hash = "sha256-PdoAOjr19aLmXYrLtMCq/tZ2Fqq7pINTuhFyMMiC0yM=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    copyDesktopItems
+  ];
+
+  desktopItems = [ (makeDesktopItem {
+    name = "TigerJython";
+    exec = "tigerjython";
+    icon = "tigerjython";
+    desktopName = "TigerJython";
+    comment     = "The Python IDE for beginners";
+    categories  = [ "Education" ];
+  }) ];
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    export JAR=$out/share/java/tigerjython2.jar
+    export CUSTOM_LIBS=$out/share/java
+    export EXAMPLES_DIR=$CUSTOM_LIBS/Examples
+
+    install -Dm444 bin/tigerjython2.jar $JAR
+    install -Dm444 bin/Lib/* --target-directory=$CUSTOM_LIBS
+    install -Dm444 bin/TestSamples/* --target-directory=$EXAMPLES_DIR
+
+    makeWrapper ${jre}/bin/java $out/bin/tigerjython \
+      --add-flags "-jar $JAR" \
+      --add-flags "-Duser.dir=$CUSTOM_LIBS/" \
+      --add-flags "-cp=.:$CUSTOM_LIBS/" \
+      --add-flags "-Xmx512M" \
+      --set _JAVA_OPTIONS '-Dawt.useSystemAAFontSettings=lcd'
+
+    runHook postInstall
+  '';
+
+  postInstall = ''
+    install -Dm444 ${icon} $out/share/icons/hicolor/64x64/apps/tigerjython.png
+  '';
+
+  # Tests need a DISPLAY
+  doCheck = false;
+
+  meta = with lib; {
```

```suggestion
  meta = {
```
Generally better to keep usage of `with` to a minimum